network/builder/src/dummy.rs

```diff
@@ -160,13 +159,15 @@ pub fn setup_network() -> DummyNetwork {
     let (listener_sender, mut listener_events) = network_builder
         .add_protocol_handler::<DummyNetworkSender, DummyNetworkEvents>(network_endpoint_config());
     network_builder.build(runtime.handle().clone()).start();
+
+    // Add the listener address with port
     let listener_addr = network_builder.listen_address();
+    seeds.insert(
```

Comment:
You also want a separate seeds struct here; you don't want to include your own peer_id/pubkey. I think the connmgr will filter out everything anyway, but it's cleaner to separate them IMO

Reply:
Connection manager does filter them out.  This just is a simple test, of if I've got all of the seeds in the network, what happens.  Works fine :P. Especially since if you're a validator, you would get your own address in discovery similarly to this.